src/store/todoSaga.ts

```diff
@@ -0,0 +1,61 @@
+import { PayloadAction } from "@reduxjs/toolkit";
+import { all, call, put, takeLatest } from "redux-saga/effects";
+import {
+  getTodosApi,
+  addTodoApi,
+  deleteTodoApi,
+  editTodoApi,
+} from "../api/todoApi";
+import {
+  Todo,
+  setTodoList,
+  addTodo,
+  deleteTodo,
+  updateTodo,
+} from "./todoSlice";
+
+function* addTodoSaga(action: PayloadAction<Todo>): Generator<any, void, any> {
+  try {
+    yield call(addTodoApi, action.payload);
+  } catch (error) {
+    console.error("Error adding todo:", error);
+  }
```

Comment:
instead of logging the error, you should have an error state in each slice and a fail action for each action. which in this case will be addTodoFail this should set the error state to true or an error message.